### PR TITLE
App finished with success, already deployed, all working correctly

### PR DESCRIPTION
**Pull request by edgardsoza**
For this project, I created a:
Splash screen
A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

Sign up and log in pages
The user should be able to register in the app with full name, email and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

Home page (categories page)
When the user logs in, they are presented with the categories page.
For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

Transactions page
For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new category" page
The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new transaction" page
The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

Testing requirements
Create unit and integration tests for all the most important components of your RoR application.

My video here:
https://www.loom.com/share/d1d90041aaeb4bf084b186b1eca5d820?sid=6496dffd-2438-45ca-b6f7-c7adcf1fbc2b

### CODE REVIEW

File: app/controllers/category_transactions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class CategoryTransactionsController < ApplicationController
```
Comment (Alfredbis29) [inline app/controllers/category_transactions_controller.rb:1]: - [ ] I was not able to access your `SessionsController` file but I noticed that the login functionality is correctly authenticating the user, but the redirection is not happening. To fix this, I think you can check if the `redirect_to` line is being executed in the create action of the SessionsController.You should ensure that there are no other lines of code or conditions that might interfere with the redirection. Also you should verify that the `categories_path` is the correct route for the desired redirection. If necessary, they can update the redirection path to the appropriate URL or path based on your application's routes and requirements. Kindly fix this , we are able to signup but not login in and navigate on other pages :wink: 

https://github.com/edgardsoza/personal-billing/assets/88894525/2c305b6c-6f5f-49bc-92f3-c989079dafb4

_This doesn't work even on the app that you deployed, kindly fix this and deploy again your app :pray:_

File: spec/features/categories_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  scenario 'User can view category details and transactions' do
+    category = Category.create(name: 'Category 1', user:, icon: 'https://example.com/category_icon.png')
+
+    visit category_path(category)
+
+    expect(page).to have_text(category.name)
+    expect(page).to have_text('0')
+    click_link 'New Transaction'
+
```
Comment (Alfredbis29) [inline spec/features/categories_spec.rb]: - [ ] I noticed that there is a syntax error in the `categories_spec.rb` file. Please  fix the error on line 23. It seems that there is an unexpected comma (,) after the `user: parameter` in the create method call. I think You should remove the comma to resolve the syntax error. Also there is another syntax error on line 23 with an unexpected closing parenthesis ()). remove this closing parenthesis as well. there is a missing `end` keyword at the end of the file. You should add the end keyword to properly close the block. As all your test should pass as required for this project kindly fix that for all your test to pass :pray:  :wink: 
Screenshot from 2023-07-13 05-48-51

**Review by Alfredbis29 — CHANGES_REQUESTED** [review]
#  STATUS: REQUIRED CHANGES 
 ## Hi @edgardsoza 
 Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

  
 ## Highlights 
:heavy_check_mark:  Your Work looks Professional 
:heavy_check_mark:  Nice work so far :+1: 
 
 ## Required Changes 

- [ ]  I just noticed this in your [Video presentation](https://www.loom.com/share/d1d90041aaeb4bf084b186b1eca5d820?sid=6496dffd-2438-45ca-b6f7-c7adcf1fbc2b) your implementation shows progress, but there are some key design elements that need attention to align with the specified requirements. To ensure consistency and a polished user experience, I kindly request the following design changes:


- Colors: Please review the chosen color scheme and ensure it matches the specified design requirements. Adjustments may be needed to achieve a cohesive and visually appealing look.

- Typography: Verify that the font face, size, and weight used throughout the application align with the design specifications. Consistency in typography will contribute to the overall design harmony. 
Kindly follow this design :point_down: 
Screenshot from 2023-07-13 06-16-23


- Layout: Pay close attention to the composition and spacing between elements. Make sure the layout adheres to the provided design guidelines, creating a balanced and intuitive user interface.

- [ ]  Check Under the review

 ## Optional suggestions 

- [ ] N/A

 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._
 
 Cheers and Happy coding! :clap: :clap: :clap:
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/category_transactions_controller.rb (hunk 1 above)
Comment (edgardsoza) [reply app/controllers/category_transactions_controller.rb:1]: I wll check this one, but that was working for me fine. I will fix it for the next review

File: spec/features/categories_spec.rb (hunk 2 above)
Comment (edgardsoza) [reply spec/features/categories_spec.rb]: solved

**Review by zainsadaqat — APPROVED** [review]
# Pull Request Approved ✅ 🎊 🎉 
## Hello @edgardsoza  👋 

### Things to appreciate
- [ ] Implemented the things requested by the previous code reviewer 👍 

### Your project is complete!  There is nothing else to say other than it's time to merge it.

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 ☑️ 



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**